### ISSUE TITLE
ListFactory seems to be broken

### ISSUE THREAD

**Issue by joaonc** [FactoryBoy/factory_boy#740]
#### Description
`ListFactory` is broken in the sorting.

`base.py:: BaseListFactory` ([here](https://github.com/FactoryBoy/factory_boy/blob/53926acc470dea7b1dbae472d659ef94074d3391/factory/base.py#L702)) the sorting is:
```
sorted(kwargs.items(), key=lambda item: int(item[0]))
```
However, the `int` is extra and breaks things.  I think it should be removed to
```
sorted(kwargs.items(), key=lambda item: item[0])
```

#### To Reproduce
Create a class that extends `ListFactory` and try to import it.
Ex:
```python
class MyFactory(ListFactory):

    class Params:
        my_param = 0

    i0 = LazyAttribute(lambda o: o.my_param + 0)
    i1 = LazyAttribute(lambda o: o.my_param + 1)
```

##### The issue
When importing, this error pops up:
```
    values = [v for k, v in sorted(kwargs.items(), key=lambda item: int(item[0]))]
E   ValueError: invalid literal for int() with base 10: 'i0'
```

I verified that by removing the conversion to `int`, the problem goes away.

#### Notes
This is new in 2.12.0.

**Comment by francoisfreitag** [comment]
I don’t think this is the intended usage of `ListFactory`. `ListFactory` is only meant to create ordered elements in a sequence, thus identified by their integer indexes. When adding the `i0` declaration, this creates a mapping where the element is identified by a string, not by its index in the sequence.

In other words, the resulting object of the `ListFactory` subclass in this example would not be a `list`.

If you can provide more details about what you are trying to achieve, I’m happy to discuss improving the library or other approaches.
